Doc/library/doctest.rst

```diff
@@ -300,33 +301,49 @@ their contained methods and nested classes.
 How are Docstring Examples Recognized?
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+The :mod:`doctest` module treats any line beginning with ``>>>`` as an 
+example to be tested. 
+Following lines which begin with ``...`` continue the example. 
+Subsequent non-blank lines, if any, form an expected output string. 
+A blank (all-whitespace) line, or a line with ``>>>`` (beginning the 
+next example), ends the expected output string.
+
 In most cases a copy-and-paste of an interactive console session works fine,
 but doctest isn't trying to do an exact emulation of any specific Python shell.
 
 ::
 
    >>> # comments are ignored
-   >>> x = 12
-   >>> x
-   12
-   >>> if x == 13:
-   ...     print("yes")
+   ... 
+   >>> import math
+   >>> x = factorial(10); math.ceil(math.log10(x))
+   7
+   >>> if x == factorial(9)*10:
+   ...     print("same:\n{0}".format(x))
    ... else:
-   ...     print("no")
-   ...     print("NO")
-   ...     print("NO!!!")
+   ...     print("differ:\n{0}\n{1}".format(x, factorial(9)*10))
    ...
-   no
-   NO
-   NO!!!
+   same:
+   3628800
    >>>
 
-Any expected output must immediately follow the final ``'>>> '`` or ``'... '``
-line containing the code, and the expected output (if any) extends to the next
-``'>>> '`` or all-whitespace line.
-
 The fine print:
 
+* The ``>>>`` string tells the module to look for an interactive statement:
+  that is, a statement list ending with a newline, or a 
+  :ref:`compound statement <compound>`. 
+  The ``...`` string tells the module that the line continues a compound
+  statement. (Actually, doctest gets these strings from the PS1 and PS2 
+  values of the :mod:`sys` module.)
+  
+* The expected output can be absent. This indicates that the example generates
+  no output when run. If the example does generate output, the module reports
+  it as a failure.
+
+* The expected output can contain multiple lines. These lines become a 
+  string, which is compared to the string of actual output from 
+  testing the example.
```

Comment:
Another point could be added:  "The last code continuation line of an example copied from the interactive shell (the line starting with "..." that is otherwise blank in the example above) may be omitted without changing the meaning of the test."